test/Microsoft.AspNet.Mvc.ModelBinding.Test/Validation/DataAnnotationsClientModelValidatorProviderTest.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    public class DataAnnotationsClientModelValidatorProviderTest
+    {
+        private readonly IModelMetadataProvider _metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+
+        [Fact]
+        public void GetValidators_DoesNotAddRequiredAttribute_ForNonNullableValueTypes_IfAttributeIsSpecifiedExplicitly()
+        {
+            // Arrange
+            var provider = new DataAnnotationsClientModelValidatorProvider();
+            var metadata = _metadataProvider.GetMetadataForProperty(typeof(DummyRequiredAttributeHelperClass),
+                                                                    "WithAttribute");
+
+            var providerContext = new ClientValidatorProviderContext(metadata);
+
+            // Act
+            provider.GetValidators(providerContext);
+
+            // Assert
+            var validator = Assert.Single(providerContext.Validators);
+            var adapter = Assert.IsType<RequiredAttributeAdapter>(validator);
```

Comment:
`GetValidators_DoesNotAddRequiredAttribute_IfAttributeIsSpecifiedExplicitly`

I think I get it. The point is that a **single** required attribute is added.